pages/api/bitbucket/callbacks/webhook.ts

```diff
@@ -0,0 +1,26 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { publishMessage } from '../../../../utils/pubsub/pubsubClient';
+import { getTopicNameFromDB } from '../../../../utils/db/relevance';
+
+const webhookHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const jsonBody = req.body;
+  const owner = jsonBody.repository.owner.username; 
+  const provider = "bitbucket";
+  const name =jsonBody.repository.name;
+  const topicName = await getTopicNameFromDB(owner, name, provider);
+  const data = {
+    'repository_provider': 'bitbucket',
+    'event_payload': jsonBody};
+  const msgType = 'webhook_callback';
+  console.info("Received bitbucket webhook event for ", name);
+  try {
+    await publishMessage(topicName, data, msgType);
```

Comment:
What happens when the message is sent successfully?

Reply:
on prem rust gets the message, does a git pull, gets pr changes, calculates hunkmap, then calls an api of next server which stores the hunkmaps in db